### ISSUE TITLE
Star Dash

### ISSUE THREAD

**Issue by TwinNumberOne** [siggame/MegaMinerAI-Game-Ideas#57]
**The** Game
- 

**TLDR**
- 
- Blow up your opponent or earn more to win
- 100x100 map plenty of room to mine and fight!
- 11 unit types, each with their own advantages and disadvantages!
- mine asteroids or planets to build ships and to earn points!

**story**
- 
The players are two company-owned AI, and their job is to secure the system for long-term mining and possible colonization. From the start of the game, one thing is clear, two competing companies are going for the same system and now the only way to complete their main objective is to destroy the other AI, and luckily for them, their main ship can produce multiple combat ship variants and some support ships. Now whos ready for some AI vs AI battle?

**Victory:**
- 
Eather the destruction of the enemy AI or the collection of the most materials in (500) turns. 

**Map:**
- 

The map is a 100x100 grid with one large asteroid belt circling the outside and it is 10 blocks from the edge and is 4 blocks thick with each asteroid holding 2000 units worth of material. It will then have two planets, and two gas giants on each side, each holding infinite units of material, with the planets needing mining stations to mine while gas giants can have both stations and ships mining it, and a class M star in the middle.

**Movement:**
- 

Each ship has a movement speed they can move that total number of squares unless they are stored in a transport in which case they move with the transport. only 1024 units worth of ships can be in any one square at any one time, but be careful as when ships attack they attack the square, not a specific unit. The mothership counts as having a cost of zero in terms of how much of a square it takes up, but only friendly units can move on that space, and any units in that space cannot attack while in said space (same with the transport ship, but not the shipyard), and it will shove any ships out of the way to make room for any new units with it choosing a random square to move them. If a ship moves into a square it shoves out an equivalent cost of units with less health then it but loses one health per unite it shoves, and it shoves it in the direction it moved in at. should a ship be shoved into a tile that is full the process starts over, should the ship be shoved into a tile with a destroyer, that ship, and the destroyer take damage equal to its remaining health (yes suicide bombing is a thing, but their AI's so don't feel bad). To clarify if a friendly ship moves into a friendly ship it is shoving, if a ship from team 1 moves into a ship from team 2 it is ramming.

**Units:**
- 

| Name | Cost | Damage |  Move Speed | Range | Health | Swivel speed |
|----|----|----|----|----|----|----|
|**Fighter Squad**          |5|1*hp|5|1|5|4|
|**Corvettes**      |4|3|4|3|3|3|
|**Turret**           |16|4|1|10|5|4|
|**Frigates**         |64|5|2|7|16|3|
|**Cruiser**          |256|7|3|4|8|2|
|**Destroyer**      |1024|8|2|7|32|1|
|**Miner**             |64|0|4|4|3|4|
|**Mining station**|256|0|1|3|8|4|
|**Ship Yard**        |256|0|1|0|8|4|
|**Mother Ship**    |only 1|0|1|0|256|0|
|**Transport**        |64|0|3|0|32|0|

Cost is measured in units of materials gathered
Damage is points of damage dealt per turn to a square
Move Speed is squares moved per turn.
Range is the range at which it can attack.
Health is well health...
Swivel speed is the speed at which a turret can turn. Turrets can face 8 directions (image of 3x3 square made of boxes with the center being the turret and the squares on the edges the barrel of the turret), where the speed is the number of directions the turret can move (left or right).
The transport can hold up to a total of 250 units worth of material, which includes ships!
Mining ships can hold up to 125 units of material, and they can mine and transport material of there mining lasers range.
Each fighter squad is composed of 5 fighters each doing 1 damage, their turret Swivel speed is tracked as a group, and can be counted as one turret, but as the squad takes damage their damage output goes down, so if the squad takes 3 damage and has 2 hp left, they can only do 2 damage.

units can go to the mother ship or shipyards to be repaired, the cost to repair is equal to their missing health.

**Combat**
- 

The main thing with combat is to make sure that your gun is facing the enemy because if so when you attack you hit, but note you don't just hit the enemy you hit their entire square (watch out for friendly fire!). Now, something to note is this, you can move and have an attack spaced around in there. This means that it is possible to have ships pop out of your transports rip an enemy apart and pop back in.

**Spawning:**
- 

The mothership can only make up to a total cost of 1024 units of ships per turn, and shipyards can make a total of 512 units of ships per turn. All ships take one turn to produce (meaning you request it during your turn and next turn you have the ships you requested). Also, should the mother ships square be full of units when the turn ticks it will then shove the unites out suffering no damage penalty.

**Comment by JeffreyStrahm** [comment]
My concerns with this game are the pure number of units. Not to mention the movement set up can cause complications. This can be fixed though.
